overlord/devicestate/devicemgr.go

```diff
@@ -435,6 +435,26 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+var startTime time.Time
+
+func init() {
+	startTime = time.Now()
+}
+
+func (m *DeviceManager) setTimeOnce(name string, t time.Time) error {
+	var prev time.Time
+	err := m.state.Get(name, &prev)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if !prev.IsZero() {
```

Comment:
`if err == ErrNoState` should be enough, unless we're worried that a zero time might get stored in the state?

Reply:
I think that was the concern yes

Reply:
if we move it to State we'll have to leave with the other semantics though. TBH I think I wrote this way just because, I probably copied some other code with that error handling and built on top.